### DISCUSSION TITLE
Error during the native iOS build suggests an issue with the code signing configuration. Xcode 16

### DISCUSSION THREAD

**Discussion by dmitriy-cr3w** [codemagic-ci-cd/codemagic-docs#2787] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

  - name: Set up provisioning profiles settings on Xcode project
        script: xcode-project use-profiles

### Expected results

Success build. I don't have this error in xcode 15.

### Actual results

Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/16-10-24_8p3i9dgr.keychain-db
/Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/models/certificate.py:88: UserWarning: Attribute's length must be >= 1 and <= 64, but it was 82
  for name_attribute in self.certificate.subject
Searching for files matching /Users/builder/clone/**/*.xcodeproj
Failed to set code signing settings for PlinksyFit.xcodeproj

Build failed :|
Step 5 script `Set up provisioning profiles settings on Xcode project` exited with status code 1

### Build id (optional)

671025307da8ce2dd98b675e

**Comment by dam-ease** [comment]
Hello @dmitriy-cr3w,
I see clean and remove steps in your build. Is there a reason you are running these?
Please can you remove the
```
xcodebuild clean -project "PlinksyFit.xcodeproj" -scheme "PlinksyFit" -configuration Release
```
and
```
rm -rf ~/Library/Developer/Xcode/DerivedData
```
steps and run the build again. If you run into any issues, please share your new buildID along with the message as well.

**Reply by Nodirbek-Khudoyberdiev** [reply]
I had the same problem with code magic xcode 16

**Reply by YazeedAlKhalaf** [reply]
i can also confirm the same problem is happening to me, xcode 16 on codemagic just now.

**Reply by dmitriy-cr3w** [reply]
@dam-ease still have problem. 
New build id: 67168dc0978b9dab4900194f

**Reply by helinanever** [reply]
Hi @Nodirbek-Khudoyberdiev and @YazeedAlKhalaf , please share your build IDs so we can look into the possible cause. Did the issue start occurring for you after upgrading to Xcode 16?

**Reply by Nodirbek-Khudoyberdiev** [reply]
Yes it started causing error after upgrading to xcode 16
here is the id of that build: 6714b94731342f707543c808
but I have already uploaded it using xcode 15
it works fine in xcode 15
@helinanever

**Reply by YazeedAlKhalaf** [reply]
sure thing @helinanever, here is the last build id: `6713d7ea243cae6cc95011d7`

i never used it before xcode 16, so can't really tell but from what i see from other here it seems that is the issue.

**Comment by priitlatt** [comment, marked as answer]
@dmitriy-cr3w @YazeedAlKhalaf @Nodirbek-Khudoyberdiev
`xcode-project use-profiles` relies on [Xcodeproj](https://github.com/CocoaPods/Xcodeproj) Ruby gem to modify Xcode project files so that specified provisioning profiles are used when building the application. Up until very recently Xcodeproj did not support projects that were generated using Xcode 16.0 or 16.1, but yesterday they [released](https://github.com/CocoaPods/Xcodeproj/releases/tag/1.26.0) an update which resolves this issue.
For the time being you can patch the issue by running
```shell
gem update xcodeproj
```
before calling `xcode-project use-profiles`.

We're working to update Xcodeproj gem directly on Codemagic VMs so that you wouldn't have to apply the update during the build.

**Reply by Dmitriy-Holovnia** [reply]
The problem with profiles is gone, but still the build is not going. Here's the build id: 6721ef084650e41832feecb9

**Reply by priitlatt** [reply]
@Dmitriy-Holovnia This looks completely unrelated issue, but I suggest you to try what is described here: https://docs.codemagic.io/troubleshooting/common-ios-issues/#error-when-using-an-adhoc-profile.
